baybe/targets/base.py

```diff
@@ -12,6 +16,9 @@
     unstructure_base,
 )
 
+if TYPE_CHECKING:
```

Comment:
Why do we do lazy import here?

Reply:
To prevent cyclic imports. `SingleTargetObjective` depends on `Objective` (which is its parent class), so `Objective` cannot have a direct dependency on `SingleTargetObjective`, which would cause the cycle